src/firstpass.rs

```diff
@@ -1183,6 +1217,120 @@ impl<'a, 'b> FirstPass<'a, 'b> {
             None
         }
     }
+
+    /// Split the usual content range and trailing attribute block.
+    ///
+    /// Returns `(leading_content_end, trailing_attribute_block)`.
+    ///
+    /// If `trailing_attribute_block` of the return value is `Some(_)`, the
+    /// string starts with `{` and ends with `}`.
+    ///
+    /// If `end_ix` is less than or equal to `begin_ix`, the given input is
+    /// considered as empty.
+    ///
+    /// The closing `}` of an attribute block can have trailing whitespaces.
+    ///
+    /// This method does not trim the trailing whitespaces. It is callers'
+    /// responsibility to trim them if necessary.
+    fn extract_attribute_block_from_header_text(
+        &self,
+        begin_ix: usize,
+        end_ix: usize,
+    ) -> (usize, Option<&'a str>) {
+        if begin_ix >= end_ix {
+            return (end_ix, None);
+        }
+
+        let bytes = self.text.as_bytes();
```

Comment:
Okay, this is again pretty nitpicky, but we are taking a `&bytes[begin_ix..some_other_ix]` subslice a few times. We can make this a bit faster by only taking one sided slices by setting `let bytes = self.text.as_bytes[begin_ix..]`. Better yet, we could just pass in `self.text.as_bytes()[begin_ix..end_ix]` and then just operate on that byte slice!